### ISSUE TITLE
Misleading errors are reported when compiler fails to infer natural type of a tuple literal

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#11291]
```
        [Fact]
        public void TargetTyping_01()
        {
            var source = @"
class C
{
    static void Main()
    {
        var x0 = new {tuple = (null, null)};
        System.Console.WriteLine(x0.ToString());
        System.ValueType x1 = (null, ""1"");
    }
}
" + trivial2uple;

            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature(), references: new[] { SystemRef });
            comp.VerifyDiagnostics(
                // (6,23): error CS0828: Cannot assign <tuple> to anonymous type property
                //         var x0 = new {tuple = (null, null)};
                Diagnostic(ErrorCode.ERR_AnonymousTypePropertyAssignedBadValue, "tuple = (null, null)").WithArguments("<tuple>").WithLocation(6, 23)
                );
        }
```

Expected: Something more specific, pointing to the actual element (even if in a nested tuple) , which is causing the inference to fail. For example:

```
                // (6,32): error CS####: Type of the tuple element cannot be infered from '<null>'.
                //         var x0 = new {tuple = (null, null)};
                Diagnostic(ErrorCode.Unknown, "null").WithArguments("<null>").WithLocation(6, 32),
                // (6,38): error CS####: Type of the tuple element cannot be infered from '<null>'.
                //         var x0 = new {tuple = (null, null)};
                Diagnostic(ErrorCode.Unknown, "null").WithArguments("<null>").WithLocation(6, 38),
```

**Comment by AlekseyTs** [comment]
A fix would require thorough testing, all places where expressions can occur syntactically should be tested.

**Comment by AlekseyTs** [comment]
@vsadov, @jcouv FYI.

**Comment by VSadov** [comment]
diagnostic refinement

**Comment by gafter** [comment]
The diagnostics look good to me now:

``` none
error CS0828: Cannot assign '(<null>, <null>)' to anonymous type property
error CS8135: Tuple with 2 elements cannot be converted to type 'ValueType'.
```

The second error is for the assignment to x1.
